aten/src/ATen/cpu/vec/vec256/vec256.h

```diff
@@ -28,29 +28,28 @@
 namespace at {
 namespace vec {
 
-// Note [Acceptable use of anonymous namespace in header]
+// Note [CPU_CAPABILITY namespace]
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-// Yes you saw right, this is an anonymous namespace in a header.  This header,
-// and all of its subheaders, REQUIRE their code to be entirely inlined into
-// the compilation unit that uses them.  It's important that these functions have
-// internal linkage so that kernels for different architectures don't get
-// combined during linking. It's sufficient to label functions "static", but
-// class methods must be an unnamed namespace to have internal linkage (since
-// static means something different in the context of classes).
-namespace {
-
- C10_UNUSED std::ostream& operator<<(std::ostream& stream, const c10::qint32& val) {
-     stream << val.val_;
-     return stream;
- }
- C10_UNUSED std::ostream& operator<<(std::ostream& stream, const c10::qint8& val) {
-     stream << static_cast<int>(val.val_);
-     return stream;
- }
- C10_UNUSED std::ostream& operator<<(std::ostream& stream, const c10::quint8& val) {
-     stream << static_cast<unsigned int>(val.val_);
-     return stream;
- }
+// This header, and all of its subheaders, will be compiled with
+// different architecture flags for each supported set of vector
+// intrinsics. So we need to make sure they aren't inadvertently
+// linked together. We do this by declaring objects in an `inline
+// namespace` which changes the name mangling, but can still be
+// accessed as `at::vec`.
+inline namespace CPU_CAPABILITY {
+
+inline std::ostream& operator<<(std::ostream& stream, const c10::qint32& val) {
```

Comment:
Can you please explain, why are you removing `C10_UNUSED` (I know that `inline` is sort of an equivalent, but `C10_UNUSED` has been validated across the plethora of compiler, so let's keep using it here)

Reply:
When the function was static, each compilation unit has it's own copy and the compiler warns that it is unused; so this macro was needed to silence the warning. But it's not needed any more  because `inline` functions are expected to not be used in every single compilation unit they're defined.